community/security/src/main/java/org/neo4j/server/security/auth/User.java

```diff
@@ -94,6 +104,7 @@ public boolean equals( Object o )
     public int hashCode()
     {
         int result = name != null ? name.hashCode() : 0;
+        result = 31 * result + ( group != null ? group.hashCode() : 0);
```

Comment:
Why would the group be part of the hashCode? That seems to imply that changing a users group somehow changes the identity of the user?